test/e2e/admin_portal.go

```diff
@@ -274,4 +274,128 @@ var _ = Describe("Admin Portal E2E Testing", func() {
 		}
 		Expect(elem.Text()).To(Equal(clusterName))
 	})
+
+	It("Should display refresh button to get latest details for each tab in cluster detail page", func() {
+		wd.Get(host + "/" + "?resourceid=" + resourceIDFromEnv())
+		wd.WaitWithTimeout(conditions.ElementIsLocated(selenium.ByID, "ClusterDetailPanel"), time.Second*3)
+
+		detailPanel, err := wd.FindElement(selenium.ByID, "ClusterDetailPanel")
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+
+		Expect(detailPanel.IsDisplayed()).To(BeTrue())
+
+		wd.Wait(conditions.ElementIsLocated(selenium.ByCSSSelector, "div[aria-label='Refresh']"))
+
+		button, err := wd.FindElement(selenium.ByCSSSelector, "div[aria-label='Refresh']")
+		button.Click()
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+
+		wd.Wait(conditions.ElementIsLocated(selenium.ByCSSSelector, "div[name='Nodes']"))
+		nodesButton, err := wd.FindElement(selenium.ByCSSSelector, "div[name='Nodes']")
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+		nodesButton.Click()
+
+		button, err = wd.FindElement(selenium.ByCSSSelector, "div[aria-label='Refresh']")
+		button.Click()
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+
+		wd.Wait(conditions.ElementIsLocated(selenium.ByCSSSelector, "div[name='Machines']"))
+		machinesButton, err := wd.FindElement(selenium.ByCSSSelector, "div[name='Machines']")
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+		machinesButton.Click()
+
+		button, err = wd.FindElement(selenium.ByCSSSelector, "div[aria-label='Refresh']")
+		button.Click()
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+
+		wd.Wait(conditions.ElementIsLocated(selenium.ByCSSSelector, "div[name='MachineSets']"))
+		machineSetsButton, err := wd.FindElement(selenium.ByCSSSelector, "div[name='MachineSets']")
+		if err != nil {
+			SaveScreenshotAndExit(wd, err)
+		}
+		machineSetsButton.Click()
```

Comment:
Can we do something like this while doing the error checks? So that we can get the required screenshot before the Click event.

Reply:
done